demos/node-classification/gcn/gcn-cora-node-classification-example.ipynb

```diff
@@ -4,14 +4,36 @@
    "cell_type": "markdown",
```

Comment:
<p>Using the ModelCheckpoint  callback to retrieve the best model is not longer necessary as EarlyStopping  has introduced the parameter <code>restore_best_weights</code>.  Note that the default values is <code>False</code>  so that it has to be explicitly set to <code>True</code>  to get the best model. Using <code>ModelCheckpoint</code>  was how we had to do it. Many of our notebooks have been updated accordingly, seems like a good opportunity to update this one as well.</p><p> </p><p>On the patience vs number of epochs decisions, e.g., reduce the former but leave the latter as is, I would suggest that we set patience to what it was and just increase the number of epochs. The point is that you want to train as much as possible to find the best model given that early stopping will help avoid overfitting. If you set patience too small, then the training might end before the optimizer had a chance to find the best model.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1125/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Reply:
<p>I agree that the old values for patience and epochs made little sense so updating them is the right thing to do. I was only suggesting that the update should have been the other way around. The value for patience should be something reasonable so there is nothing special about it being 50. Generally, it is preferable to train as long as possible but also give the algorithm a fair chance to overcome a local minimum, hence generally you want a largish patience value but not too large with respect to the number of epochs as we had it setup before. I think we had set the number of epochs to a small number so that it runs fast; not sure why patience was so high though, probably an oversight. </p><p> </p><p> </p>